c_binding/lsm_mgmt.cpp

```diff
@@ -2653,3 +2653,66 @@ int lsm_battery_list(lsm_connect *c, const char *search_key,
     rc = rpc(c, "batteries", parameters, response);
     return get_battery_array(c, rc, response, bs, count);
 }
+
+int lsm_volume_cache_info(lsm_connect *c, lsm_volume *volume,
+                          uint32_t *write_cache_policy,
+                          uint32_t *write_cache_status,
+                          uint32_t *read_cache_policy,
+                          uint32_t *read_cache_status,
+                          uint32_t *physical_disk_cache, lsm_flag flags)
+{
+    if (LSM_FLAG_UNUSED_CHECK(flags)) {
+        return LSM_ERR_INVALID_ARGUMENT;
+    }
+
+    int rc = LSM_ERR_OK;
+    CONN_SETUP(c);
+
+    if (!LSM_IS_VOL(volume)) {
+        rc = LSM_ERR_INVALID_ARGUMENT;
+        goto out;
+    }
+
+    if (write_cache_policy == NULL || write_cache_status == NULL ||
+        read_cache_policy == NULL || read_cache_status == NULL ||
+        physical_disk_cache == NULL) {
+
+        rc = LSM_ERR_INVALID_ARGUMENT;
+        goto out;
+    }
```

Comment:
I know people like having a single exit point, but this early in the validation process I'm thinking it would be easier and cleaner to just return here and above with the invalid argument.  That would alleviate the need for having all the if checks below when clearing all the `"out"` variables.